src/core/registration.c

```diff
@@ -46,6 +52,36 @@ MsQuicRegistrationOpen(
         QUIC_TRACE_API_REGISTRATION_OPEN,
         NULL);
 
+    if (ExternalRegistration) {
```

Comment:
First, let's move this after the invalid parameter check below. Second, internal registration is always **after** you create an external one, so the Datapath MUST have already been initialized. You should `CXPLAT_DBG_ASSERT` that and clean up how/when you do these checks. For instance, no need to check in the failure path below when releasing the lock.

Reply:
Addressed in the latest commit.